### PR TITLE
Improve gas price estimator

### PR DESCRIPTION
**Pull request by magik6k**
Definitely very far from the best, but it should make it slightly smarter now

Results for calibration network as of now:
```
$ ./lotus chain gas-price
1 blocks: 1250001 (0.000000000001250001 FIL)
2 blocks: 1250001 (0.000000000001250001 FIL)
3 blocks: 1250001 (0.000000000001250001 FIL)
5 blocks: 1250001 (0.000000000001250001 FIL)
10 blocks: 10 (0.00000000000000001 FIL)
20 blocks: 1250001 (0.000000000001250001 FIL)
50 blocks: 10 (0.00000000000000001 FIL)
100 blocks: 1 (0.000000000000000001 FIL)
300 blocks: 1 (0.000000000000000001 FIL)
```

### CODE REVIEW

File: node/impl/full/gas.go [hunk 1]
```diff
@@ -26,11 +28,75 @@ const MinGasPrice = 1
...
+			continue
+		}
+
+		return price.price, nil
+	}
+
 	switch nblocksincl {
-	case 0:
-		return types.NewInt(MinGasPrice + 2), nil
 	case 1:
```
Comment (Kubuxu) [inline node/impl/full/gas.go:97]: Remove the old code?

**Comment by Kubuxu** [comment]
And we probably want to go around and change GasPrice to 0.

File: node/impl/full/gas.go (hunk 1 above)
Comment (magik6k) [reply node/impl/full/gas.go:97]: We still may want this to set some price when there are no messages

Comment (Kubuxu) [reply node/impl/full/gas.go:97]: Aah.

File: api/test/mining.go [hunk 2]
```diff
@@ -26,14 +26,18 @@ func (ts *testSuite) testMining(t *testing.T) {
 	ctx := context.Background()
 	apis, sn := ts.makeNodes(t, 1, oneMiner)
 	api := apis[0]
+	//minerAddr, _ := address.NewIDAddress(1000)
```
Comment (whyrusleeping) [inline api/test/mining.go:29]: rm comment?

**Pull request merged**